server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -929,7 +929,7 @@ private AllocateUnassignedDecision decideAllocateUnassigned(final ShardRouting s
                             updateMinNode = ((((nodeHigh > repId && minNodeHigh > repId)
                                                    || (nodeHigh < repId && minNodeHigh < repId))
                                                   && (nodeHigh < minNodeHigh))
-                                                 || (nodeHigh > minNodeHigh && nodeHigh > repId && minNodeHigh < repId));
```

Comment:
why is this ok?

Reply:
`nodeHigh > minNodeHigh == true` is implied by the line above `nodeHigh < minNodeHigh == false` (somehow I feel we should maybe indent everything above this line a little differently, it's kinda horrible to read with different levels of precedence at the same indent level imo).